api/query/search.go

```diff
@@ -126,8 +126,17 @@ func (sh structuredSearchHandler) ServeHTTP(w http.ResponseWriter, r *http.Reque
 		return
 	}
 
-	simpleQ, ok := rq.AbstractQuery.(TestNamePattern)
-	if !ok {
+	// Check if the query is a simple (empty, or test name only) query
+	var simpleQ TestNamePattern
+	var isSimpleQ bool
+	{
```

Comment:
Why the scoping?

Reply:
No major reason; avoids variable pollution.

Reply:
Doesn't seem necessary to me, but no strong opinion.